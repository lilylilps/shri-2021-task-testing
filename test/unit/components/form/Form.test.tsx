import React from "react";
import renderer from 'react-test-renderer';

import { it, describe, expect } from "@jest/globals";

import { Form } from "../../../../src/client/components/Form";
import { render, fireEvent } from "@testing-library/react";
import events from "@testing-library/user-event";

describe('проверка компонента form', () => {
    const onSubmit = jest.fn();

    it('snapshot', () => {
        const tree = renderer.create(<Form onSubmit={onSubmit} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('валидация проходит', () => {
        const form = <Form onSubmit={onSubmit} />

        const { getByRole, getByLabelText } = render(form);

        const nameInput = getByLabelText('Name');
        fireEvent.change(nameInput, {target: {value: 'lil'}});

        const phoneInput = getByLabelText('Phone');
        fireEvent.change(phoneInput, {target: {value: '83344433333'}});

        const addressInput = getByLabelText('Address');
        fireEvent.change(addressInput, {target: {value: 'mosc'}});

        const submitButton = getByRole('button', { name: /checkout/i });
        events.click(submitButton);

        expect(onSubmit).toHaveBeenCalled();
    })
});