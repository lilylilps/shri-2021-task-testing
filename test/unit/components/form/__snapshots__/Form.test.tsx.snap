// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка компонента form snapshot 1`] = `
<div
  className="Form"
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-name"
    >
      Name
    </label>
    <input
      autoComplete="off"
      className="Form-Field Form-Field_type_name form-control"
      disabled={false}
      id="f-name"
      onChange={[Function]}
      type="text"
    />
    <div
      className="invalid-feedback"
    >
      Please provide your name
    </div>
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-phone"
    >
      Phone
    </label>
    <input
      className="Form-Field Form-Field_type_phone form-control"
      disabled={false}
      id="f-phone"
      onChange={[Function]}
      type="text"
    />
    <div
      className="invalid-feedback"
    >
      Please provide a valid phone
    </div>
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-address"
    >
      Address
    </label>
    <textarea
      className="Form-Field Form-Field_type_address form-control"
      disabled={false}
      id="f-address"
      onChange={[Function]}
      rows={3}
    />
    <div
      className="invalid-feedback"
    >
      Please provide a valid address
    </div>
  </div>
  <button
    className="Form-Submit btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;
